hazelcast/src/main/java/com/hazelcast/query/PagingPredicate.java

```diff
@@ -71,16 +70,13 @@
  */
 public class PagingPredicate implements IndexAwarePredicate, DataSerializable {
 
-    private Predicate predicate;
+    private static final Map.Entry<Integer, Map.Entry> NULL_ANCHOR = new SimpleImmutableEntry(-1, null);
 
+    private final LinkedList<Map.Entry<Integer, Map.Entry>> anchorList = new LinkedList<Map.Entry<Integer, Map.Entry>>();
```

Comment:
Yes, that's what we assume (should be always valid)
we can add some asserts for that